workflows/illuminaNcov.nf

```diff
@@ -91,7 +92,13 @@ workflow sequenceAnalysis {
       readMapping(readTrimming.out.combine(ch_preparedRef))
 
       trimPrimerSequences(readMapping.out.combine(ch_bedFile))
- 
+
+     // uncomment to generate cram for mapped bams before primer trimming step...     
+     // bamToCram(trimPrimerSequences.out.mapped.combine(ch_preparedRef.map{ it[0] }))
+
+     // uncomment to generate cram for mapped and primer trimmed bams ...     
+      bamToCram(trimPrimerSequences.out.ptrim.combine(ch_preparedRef.map{ it[0] }))
```

Comment:
Rather than have these 2 ways of doing it with 1 commented out, I think you should follow what is done to provide qc passed bams to collateSamples(). Calling bamToCram should also only happen when an --output_cram option is supplied by the user.

Reply:
update this now